eZ/Publish/Core/Persistence/Cache/ObjectStateHandler.php

```diff
@@ -251,19 +278,22 @@ public function setContentState($contentId, $groupId, $stateId)
      */
     public function getContentState($contentId, $stateGroupId)
     {
-        $cacheItem = $this->cache->getItem('ez-state-by-group-' . $stateGroupId . '-on-content-' . $contentId);
-        if ($cacheItem->isHit()) {
-            return $cacheItem->get();
-        }
-
-        $this->logger->logCall(__METHOD__, ['contentId' => $contentId, 'stateGroupId' => $stateGroupId]);
-        $contentState = $this->persistenceHandler->objectStateHandler()->getContentState($contentId, $stateGroupId);
-
-        $cacheItem->set($contentState);
-        $cacheItem->tag(['state-' . $contentState->id, 'content-' . $contentId]);
-        $this->cache->save($cacheItem);
-
-        return $contentState;
+        return $this->getCacheValue(
+            (int) $stateGroupId,
+            'ez-state-by-group-',
+            function (int $stateGroupId) use ($contentId): ObjectState {
+                $this->logger->logCall(__METHOD__, ['contentId' => (int) $contentId, 'stateGroupId' => $stateGroupId]);
+
+                return $this->persistenceHandler->objectStateHandler()->getContentState((int) $contentId, (int) $stateGroupId);
```

Comment:
That type cast is redundant, since the function already accepts `int`s only.
```suggestion
                return $this->persistenceHandler->objectStateHandler()->getContentState((int) $contentId, $stateGroupId);
```